components/script/layout_wrapper.rs

```diff
@@ -783,15 +759,16 @@ impl<'ln> ServoThreadSafeLayoutNode<'ln> {
 
 impl<'ln> NodeInfo for ServoThreadSafeLayoutNode<'ln> {
     fn is_element(&self) -> bool {
-        if let Some(LayoutNodeType::Element(_)) = self.type_id() { true } else { false }
+        self.pseudo == PseudoElementType::Normal && self.node.is_element()
     }
 
     fn is_text_node(&self) -> bool {
-        if let Some(LayoutNodeType::Text) = self.type_id() { true } else { false }
+        self.pseudo == PseudoElementType::Normal && self.node.is_text_node()
```

Comment:
Oh, right? I thought we treated pseudo-elements as elements, but you can be right.